### PR TITLE
Test skeletons for chain submodule - Related to #792

### PR DESCRIPTION
**Pull request by Tschakki**
related to #792

### CODE REVIEW

File: test/unit/modules/blocks/chain.js [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+		describe('when library.db.tx callback does not throw', function () {
+			
+			it('should call __private.afterSave with block');
+			
+			it('should call __private.afterSave with callback');
+		});
+	});
+	
+	describe('__private', function () {
+		describe('afterSave', function () {
```
Comment (Isabello) [inline test/unit/modules/blocks/chain.js]: space

File: test/unit/modules/blocks/chain.js [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+					it('should set genesis block as last block');
+					
+					it('should call callback');
+				});
+			});
+		});
+	});
+	
+	describe('__private', function () {
+		describe('applyTransaction', function () {
```
Comment (Isabello) [inline test/unit/modules/blocks/chain.js]: space

File: test/unit/modules/blocks/chain.js [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+		
+		it('should execute library.bus.message with reducedBlock');
+		
+		it('should execute library.bus.message with broadcast');
+		
+		it('should execute library.logger.debug with blockId');
+	});
+	
+	describe('__private', function () {
+		describe('popLastBlock', function () {
```
Comment (Isabello) [inline test/unit/modules/blocks/chain.js]: space

File: test/unit/modules/blocks/utils.js [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+					it('should call log');					
+				});
+			});
+			
+			describe('log function', function () {
+				
+				it('should call library.logger.info');
+				
+				it('should call library.logger.info with msg');				
+				it('should call library.logger.info with message');
```
Comment (Isabello) [inline test/unit/modules/blocks/utils.js]: space

File: test/unit/modules/blocks/chain.js [hunk 5]
```diff
@@ -0,0 +1,636 @@
...
+				it('should call self.saveBlock with library.genesisblock.block');
+				
+				describe('when self.saveBlock fails', function () {
+					
+					it('should call callback with error');
+				});
+				
+				describe('when self.saveBlock succeeds', function () {
+					
+					it('should call callback with error = undefined');
```
Comment (MaciejBaj) [inline test/unit/modules/blocks/chain.js]: how about the result here?

Comment (MaciejBaj) [inline test/unit/modules/blocks/chain.js]: how about the result here?

File: test/unit/modules/blocks/chain.js [hunk 6]
```diff
@@ -0,0 +1,636 @@
...
+				it('should call self.saveBlock');
+
+				it('should call self.saveBlock with library.genesisblock.block');
+				
+				describe('when self.saveBlock fails', function () {
+					
+					it('should call callback with error');
+				});
+				
+				describe('when self.saveBlock succeeds', function () {
```
Comment (MaciejBaj) [inline test/unit/modules/blocks/chain.js]: the block not needed - the same action will take place also for if block id will be present

File: test/unit/modules/blocks/chain.js [hunk 7]
```diff
@@ -0,0 +1,636 @@
...
+			it('should call logger.error');
+			
+			it('should call logger.error with err.stack');
+
+			it('should call callback with "Blocks#saveGenesisBlock error"');
+		});
+
+		describe('when db query succeeds', function () {
+
+			describe('and blockId is undefined', function () {
```
Comment (MaciejBaj) [inline test/unit/modules/blocks/chain.js]: 'when genesis block does not exist'

File: test/unit/modules/blocks/chain.js [hunk 8]
```diff
@@ -0,0 +1,636 @@
...
+					it('should call callback with error');
+				});
+				
+				describe('when self.saveBlock succeeds', function () {
+					
+					it('should call callback with error = undefined');
+				});
+			});
+
+			describe('and blockId is defined', function () {
```
Comment (MaciejBaj) [inline test/unit/modules/blocks/chain.js]: remove the context and define the behaviour as common

File: test/unit/modules/blocks/chain.js [hunk 9]
```diff
@@ -0,0 +1,636 @@
...
+				it('should call callback with error = undefined');
+				
+				it('should call callback with result = undefined');
+			});
+		});
+	});
+
+	describe('saveBlock', function () {
+
+		describe('call library.db.tx', function () {
```
Comment (MaciejBaj) [inline test/unit/modules/blocks/chain.js]: try instead:
```
it('should call library.db.tx')
it('should call library.logic.block.dbSave');
it('should call library.logic.block.dbSave with block');
```

File: test/unit/modules/blocks/chain.js [hunk 10]
```diff
@@ -0,0 +1,636 @@
...
+			});
+		});
+	});
+
+	describe('applyBlock', function () {
+
+		it('should call modules.blocks.isActive');
+		
+		it('should call modules.blocks.isActive with true');
+		//undoUnconfirmedList
```
Comment (MaciejBaj) [inline test/unit/modules/blocks/chain.js]: please remove all the comments

File: test/unit/modules/blocks/chain.js [hunk 11]
```diff
@@ -0,0 +1,636 @@
...
+							
+							it('should call modules.transactions.undo with oldLastBlock');
+							
+							it('should call modules.transactions.undo with sender');
+							
+							it('should call modules.transactions.undoUnconfirmed');
+							
+							it('should call modules.transactions.undoUnconfirmed with transaction');
+							
+							//after loop through oldLastBlock.transactions
```
Comment (MaciejBaj) [inline test/unit/modules/blocks/chain.js]: not necessary

File: test/unit/modules/blocks/chain.js [hunk 12]
```diff
@@ -0,0 +1,620 @@
+'use strict';
+
+describe('blocks/chain', function () {
+
+	describe('Chain constructor', function () {
```
Comment (4miners) [inline test/unit/modules/blocks/chain.js]: just `constructor`

File: test/unit/modules/blocks/chain.js [hunk 13]
```diff
@@ -0,0 +1,620 @@
...
+
+			it('should assign db');
+
+			it('should assign genesisblock');
+
+			it('should assign bus');
+
+			it('should assign balanceSequence');
+
+			describe('should assign logic',function () {
```
Comment (4miners) [inline test/unit/modules/blocks/chain.js]: missing space

**Pull request merged**